app/controllers/WhatDoYouWantToDoController.scala

```diff
@@ -120,12 +118,18 @@ class WhatDoYouWantToDoController @Inject()(
 
   private def allowWhatDoYouWantToDo(implicit request: Request[AnyContent], user: AuthedUser): Future[Result] = {
     val nino = user.nino
-    for {
-      hasTaxCodeChanged <- taxCodeChangeService.hasTaxCodeChanged(nino)
-      showJrsTile       <- jrsService.checkIfJrsClaimsDataExist(nino)
-      model             <- whatToDoView(nino, hasTaxCodeChanged, showJrsTile)
-      _                 <- auditNumberOfTaxCodesReturned(nino, showJrsTile)
-    } yield Ok(whatDoYouWantToDoTileView(WhatDoYouWantToDoForm.createForm, model, applicationConfig))
+
+    taxCodeChangeService.hasTaxCodeChanged(nino) flatMap {
```

Comment:
This just handles taxCodeChangeService.hasTaxCodeChanged returning an error, what if the others in the for loop return an error?

Reply:
Yes they can so we have to handle each exception being thrown around by code